modules/ocf_printhost/files/ocfps

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Use the user command pdftops, not the CUPS filter.
+PDFTOPS=/usr/bin/pdftops
+
+if [ "$#" -ne 5 ] && [ "$#" -ne 6 ]; then
+    echo "Usage: $0 job-id user title copies options [file]"
+    exit
+fi
+
+if [ "$#" -eq 5 ]; then
+    $PDFTOPS - - | pstops "$@"
+else
+    $PDFTOPS "$6" - | pstops "${@:1:5}"
+fi
```

Comment:
Of course. `ocfps` is a CUPS filter, and all CUPS filters have to have the same signature as in the usage string above. If the 6th arg is present, then the filter should take job data (the document to print) from the given file. Otherwise, the filter takes data from stdin. In either case, `ocfps` runs the actual document through `pdftops`, a rasterizing converter provided by the `poppler-utils` package (this is unrelated to the CUPS filter also named `pdftops`). Finally, this modified document is passed to the CUPS filter `pstops` along with the unaltered environment vars and remaining command line arguments provided by cups.

`pstops` is a CUPS filter located in `/usr/lib/cups/filter/`, which is added to the PATH by CUPS. Originally, I specified the full path in a variable `PSTOPS`, but undid that during debugging in case CUPS was adding a different location to the PATH. I can redefine the variable to clarify if that seems useful.

This explanation will be in the documentation.